packages/store/addon/-private/system/model/internal-model.ts

```diff
@@ -430,31 +412,6 @@ export default class InternalModel {
 
           this._record = store._modelFactoryFor(this.modelName).create(createOptions);
           setRecordIdentifier(this._record, this.identifier);
-          if (DEBUG) {
-            let klass = this._record.constructor;
-            let deprecations = lookupDeprecations(klass);
-            [
-              'becameError',
-              'becameInvalid',
-              'didCreate',
-              'didDelete',
-              'didLoad',
-              'didUpdate',
-              'ready',
-              'rolledBack',
-            ].forEach(methodName => {
-              if (this instanceof Model && typeof this._record[methodName] === 'function') {
```

Comment:
`this` here was always an instance of `InternalModel` so this conditional was always `false`